modules/disk-uploader/pkg/vmexport/vmexport.go

```diff
@@ -99,6 +99,43 @@ func GetRawDiskUrlFromVolumes(client kubecli.KubevirtClient, namespace, name, vo
 	return "", fmt.Errorf("volume %s is not found in VirtualMachineExport internal volumes", volumeName)
 }
 
+func GetLabelsFromExportSource(k8sClient *k8sv1.CoreV1Client, virtClient kubecli.KubevirtClient, exportSourceKind, exportSourceNamespace, exportSourceName, volumeName string) (map[string]string, error) {
+	source, err := getExportSource(exportSourceKind, exportSourceName)
+	if err != nil {
+		return nil, err
+	}
+
+	var sourceLabels map[string]string
+	switch source.Kind {
+	case "VirtualMachine", "VirtualMachineSnapshot":
+		dv, err := virtClient.CdiClient().CdiV1beta1().DataVolumes(exportSourceNamespace).Get(context.Background(), volumeName, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		sourceLabels = dv.GetLabels()
+	case "PersistentVolumeClaim":
+		pvc, err := k8sClient.PersistentVolumeClaims(exportSourceNamespace).Get(context.Background(), source.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		sourceLabels = pvc.GetLabels()
+	default:
+		return nil, fmt.Errorf("unsupported source kind: %s", source.Kind)
+	}
+
+	return filterLabels(sourceLabels), nil
+}
+
+func filterLabels(labels map[string]string) map[string]string {
```

Comment:
This isn't required.

Reply:
Why not? I want to make sure that it returns only required labels and not all the labels.

Reply:
Seems a little redundant to filter when the code extracting the label values within `DefaultConfig` would work without this?